tests/ludwig/schema/test_schema_fields.py

```diff
@@ -0,0 +1,27 @@
+from ludwig.constants import ALLOF, ENCODER, ENUM, INPUT_FEATURES, ITEMS, PROPERTIES, THEN
+
+
+def parse_schema(input_dict: dict, key_list: list):
+    """Parse a nested dictionary by a list of keys."""
+    parsed_dict = input_dict
+    for key in key_list:
+        parsed_dict = parsed_dict[key]
+    return parsed_dict
+
+
+def test_transformer_encoder_representations(ecd_schema):
```

Comment:
I'm not convinced that we absolutely need this test, since it seems like it's testing implementation details, and we already have tests for `StringOptions`.

Guidance on not testing implementation details:
https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

Reply:
Ya I feel like it's a little pedantic, however Travis asked specifically for this test [here](https://docs.google.com/spreadsheets/d/1By3vMX17dVwpvs0lyiLim5su4gp7pHlA3L7542WtbaI/edit?disco=AAAApOfcHF4)
```
Travis Addair
12:36 PM Yesterday
And add unit test (enumerate all possible values, check does not allow None).
```
So I interpreted that as I should build this test. Happy to remove though if we feel it's unnecessary.

Reply:
It looks like there's already a test for `allow_none=False` for `StringOptions`.

https://github.com/ludwig-ai/ludwig/blob/master/tests/ludwig/marshmallow/test_fields_misc.py#L20-L34

I'd say we can do without this test.

Reply:
Sounds good, will remove!